### ISSUE TITLE
Matching Pair Movement With vimtex-% Not Added To Jumplist

### ISSUE THREAD

**Issue by hwagyesa** [lervag/vimtex#479]
The remapped `%` functionality in vimtex doesn't add the cursor position prior to jumping to the matching pair to the jumplist. The relevant code in vimtex#motion#find_matching_pair is:

```
call cursor(delim.lnum,
        \ (delim.is_open
        \   ? delim.cnum
        \   : delim.cnum + strlen(delim.match) - 1))
```

This breaks a common workflow for deleting enclosing matching pairs that aren't vimtex-specific, such as simple braces around the arguments of a `\begin{...}` statement. One might typically delete these by positioning the cursor on the first brace and entering ```%x``x```, but with the remapped `%` key not changing the jumplist this isn't possible. 

A quick fix (source: selected answer [here](http://stackoverflow.com/questions/19195160/push-a-location-to-the-jumplist)) would be to use something like this instead of the code above:

```
  execute "normal " . delim.lnum . "G" . (delim.is_open ? delim.cnum :
        \ delim.cnum + strlen(delim.match) - 1) . "|"
```

This moves the cursor as expected and adds the previous cursor position to the jumplist.

I haven't listed a MWE since entering any tex document and testing this on the `\documentclass{...}` line should suffice. I tested the non-vimtex behavior by checking the output of `:map` in the `motion.vim` file and using `%` on some matching parentheses.

Sorry if I haven't followed accepted practices with this posting. I am new to Github, but I have enjoyed using this plugin and hoped to contribute something that had come up in my daily use. Thank you!

**Comment by lervag** [comment]
Thanks. I wasn't aware that this was the common behaviour. I'll look into it.

**Comment by hwagyesa** [comment]
After investigating this a bit more, I observed that when the first non-whitespace character in a line is part of a matching pair with the matching delimiter on the same line of text, using ```%`` ``` with the cursor on the opening delimiter fails to return to the cursor's original position if the command 

```
execute "normal " . delim.lnum . "G" . (delim.is_open ? delim.cnum :
        \ delim.cnum + strlen(delim.match) - 1) . "|"
```

is used in `vimtex#motion#find_matching_pair`, as I suggested above.

Using `:ju` to check the most recent jump and `:marks` to check the line/column of the `'` mark after keying `%` on the opening delimiter, as in the test setup described above, shows that although the jumplist registers the old position, a mark is not registered and so ``` `` ``` does not return to the start of the line (although `<C-o>` does return properly). I could not figure out why this occurs. In `motion.txt` the explanation given for the ``` `` ``` command is

```
''  ``			To the position before the latest jump, or where the
			    last "m'" or "m`" command was given. 
```

The "or" seems somewhat ambiguous in this situation. Using `G<#1>|<#2>`, with `<#1>` as the line number and `<#2>` as the column number, outside of this case leaves a mark at the pre-jump location and registers it in the jumplist as expected. I expect there may be a simple reason for this and I apologize that I couldn't present it myself.

Testing in the motion.vim file (i.e. `%` is not remapped) with two matching delimiters on the same line with the opening delimiter being the first non-whitespace character on the line shows the expected behavior of `<C-o>` and ``` `` ``` both returning to the opening delimiter after keying `%` with the cursor on the opening delimiter. 

Finally, contrary to the suggestion I provided above, it seems that keeping the original `cursor(...)` call in the `motion.vim` file but manually adding a mark prior (i.e. with ``normal! m` ``) gives the expected behavior with both `<C-o>` and ``` `` ``` under the conditions described above. This was also suggested in the post I linked to previously ([here](http://stackoverflow.com/questions/19195160/push-a-location-to-the-jumplist)).
